utils/mg/patches/900-musl_compatibility.patch

```diff
@@ -0,0 +1,16 @@
+--- a/fileio.c
++++ b/fileio.c
+@@ -23,6 +23,13 @@
+ #include <string.h>
+ #include <unistd.h>
+ 
++#ifndef DEFFILEMODE
++#define DEFFILEMODE (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)
++#endif
++#ifndef MAXNAMLEN
```

Comment:
A quick check shows musl defining this in sys/param.h and dirent.h for uclibc-ng.

Reply:
A closer look shows MAXNAMLEN to be a BSD define whereas the POSIX one is NAME_MAX.

Reply:
musl : DEFFILEMODE is undefined, MAXNAMLEN is define in sys/param.h
uclibc:  DEFFILEMODE is definedin  sys/stat.h, MAXNAMLEN is defined in dirent.h

Anyway, my patch works well in both environments.

Reply:
Hmm? I git grepped the codebase and found only one usage of MAXNAMLEN in some DragonflyBSD header. It seems MAXNAMLEN was already replaced by NAME_MAX in fileio.c.

Reply:
Yeah. https://github.com/ibara/mg/commit/bc00df55d725b3a1e843003afb17d24dbaf4d23f

Reply:
Oh. I did not know that MAXNAMLEN was not used in the new version.